mmeds/database/database.py

```diff
@@ -247,6 +247,57 @@ def set_mmeds_user(self, user):
         self.db.commit()
         return set_user
 
+    def extract_columns_from_where_query(self, where):
+        """
+        Given an SQL WHERE statement, extract only the relevant columns
+        Currently does not support the operators "BETWEEN", "LIKE", or "IN"
+        """
+        keyword_delimiter = r"\s(?:AND|and|OR|or)\s"
+        column_separator = r"(?:['\"`]?)([a-zA-Z0-9_\-]+)(?:['\"`]?\s*)(?:[<>!]?=|[<>])(?:\s*[a-zA-Z0-9'\"`]+)"
+
+        split_where = re.split(keyword_delimiter, where)
+        columns = []
+        for w in split_where:
+            w = w.strip()
+            matches = re.findall(column_separator, w)
+            if not len(matches) == 1:
+                raise ValueError(f"Could not resolve '{w}' to a single where clause")
+            columns.append(matches[0])
+        return columns
+
+    def query_meta_analysis(self, where):
+        """
+        Execute a query to get the full set of studies, SampleIDs, and metadata
+        paths that match the given query
+        =========================================================
+        :where: An SQL-formatted WHERE statement for matching to the desired samples
+        """
+        # Extract the relevant column names
+        columns = self.extract_columns_from_where_query(where)
+        Logger.debug(f"where columns: {columns}")
+        columns_str = ""
+        # Add the relevant columns to the query, RawDataID and StudyName are present by default
+        for c in columns:
+            if not c == 'RawDataID' and not c == 'StudyName':
+                columns_str += f", `{c}`"
+        data, header = self.execute(fmt.SELECT_META_ANALYSIS_QUERY.format(columns=columns_str, where=where))
+
+        # Only collect the relevant RawDataIDs, organized by StudyName
+        ret_entries = {}
+        for entry in data:
```

Comment:
couldn't you just read all the data into a dataframe and then drop duplicates?
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.drop_duplicates.html